### PR TITLE
HPCC-31963 Improve LDAP server initialization failure log messages

### PR DESCRIPTION
**Pull request by kenrowland**
Added additional information to log messages issued when initial connection to the Active Directory fails during initialization

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/security/LdapSecurity/ldaputils.cpp [hunk 1]
```diff
@@ -302,12 +302,28 @@ int LdapUtils::getServerInfo(const char* ldapserver, const char* userDN, const c
...
+                else
+                {
+                    DBGLOG("Unable to connect. If you're trying to connect to an LDAPS server, you may need to specify a different cipher suite using the 'ldapCipherSuite' attribute in the LDAP configuration.");
+                }
+            }
 
-        if(nullptr == ld)
             return err;//unable to connect, give up
+        }
+
```
Comment (jakesmith) [inline system/security/LdapSecurity/ldaputils.cpp:326]: trivial: spurious extra line

File: system/security/LdapSecurity/ldaputils.cpp [hunk 2]
```diff
@@ -302,12 +302,28 @@ int LdapUtils::getServerInfo(const char* ldapserver, const char* userDN, const c
...
         }
 
-        // for new versions of openldap, version 2.2.*
-        if(nullptr == ld   &&  err == LDAP_PROTOCOL_ERROR  &&  stype != ACTIVE_DIRECTORY)
-            DBGLOG("If you're trying to connect to an OpenLdap server, make sure you have \"allow bind_v2\" enabled in slapd.conf");
+        if (ld == nullptr)
+        {
+            if (err == LDAP_PROTOCOL_ERROR && stype != ACTIVE_DIRECTORY)
+            {
+                DBGLOG("Unable to connect. If you're trying to connect to an OpenLdap server, make sure you have \"allow bind_v2\" enabled in slapd.conf");
```
Comment (jakesmith) [inline system/security/LdapSecurity/ldaputils.cpp:309]: DBBLOG will not be seen in some systems that are defaulting to logging level of 55 (Progress level messages and above only).
I suspect this/others should be a WARNLOG's

**Review by jakesmith — COMMENTED** [review]
@kenrowland - one comment re. logging level.

File: system/security/LdapSecurity/ldaputils.cpp (hunk 2 above)
Comment (kenrowland) [reply system/security/LdapSecurity/ldaputils.cpp:309]: I was not comfortable with DBGLOG for the same reason, but was unsure what to use. Thanks for the suggestion.

**Review by jakesmith — APPROVED** [review]
@kenrowland - looks fine. Please squash.

**Comment by kenrowland** [comment]
@ghalliday Please merge

**Pull request merged**